drivers/include/ds1307.h

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_ds1307  DS1307 RTC
+ * @ingroup     drivers_sensors
+ * @brief       Device drive interface for the DS1307 real-time clock
+ * @{
+ *
+ * @file
+ * @brief       DS1307 definitions
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef DS1307_H
+#define DS1307_H
+
+#include <stdbool.h>
+#include <time.h>
+
+#include "nvram.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   I2C address of DS1307 RTC
+ */
+#ifndef DS1307_I2C_ADDRESS
+#define DS1307_I2C_ADDRESS      (0x68)
+#endif
+
+/**
+ * @brief   Maximum I2C bus speed to use with the device
+ */
+#ifndef DS1307_I2C_MAX_CLK
+#define DS1307_I2C_MAX_CLK      (I2C_SPEED_FAST)
+#endif
+
+/**
+ * @brief   Maximum size in byte of on-chip NVRAM
+ */
+#define DS1307_NVRAM_MAX_SIZE   (56)
+
+/**
+ * @brief   Device descriptor for DS1307 devices
+ */
+typedef struct {
+    i2c_t i2c;                  /**< I2C bus the device is connected to */
```

Comment:
Then a lot of other `param` structs for I2C are false too. I don't think that would be correct.